dvc/remote/base.py

```diff
@@ -691,7 +691,7 @@ def path_to_checksum(self, path):
     def checksum_to_path_info(self, checksum):
         return self.path_info / checksum[0:2] / checksum[2:]
 
-    def list_cache_paths(self, prefix=None):
+    def list_cache_paths(self, prefix=None, progress_callback=None):
```

Comment:
`progress_callback` is unused in ssh and local remotes, but the parameter was added to both so that overridden `list_cache_paths()` in ssh/local still match the abstract method defined in `RemoteBASE.list_cache_paths()`

Reply:
yes but presumably it should be used now that it's added?

Reply:
both local and ssh have their own overridden `cache_exists()` implementation and neither one uses `list_cache_paths()` at all during `cache_exists()`. The only time `list_cache_paths()` would get called for either local or ssh remotes would be for `RemoteBASE.all()` during a `dvc gc` call.

`all()` does not currently display any progressbars at all, but now that I'm thinking about this, it probably should

Reply:
indeed; though I'm not sure in principle about having a function exists which takes a non-null `progress_callback` and ignores it. This implies a bar will be displaying hanging at 0% on screen and then eventually suddenly disappear.

Reply:
`list_cache_paths()` for local and ssh have been updated to use `progress_bar` (if it is set) to keep them consistent with the other remotes. A separate issue (#3543) has been filed for updating the `dvc gc`/`RemoteBASE.all()` behavior and will be addressed in a follow-up PR.

Reply:
btw just to clarify - every time I've said "unused?" it's more a reminder to address at some point (could be in a future PR) rather than an expectation to address in this PR.